src/App/Order/Calculator/PostNl/PostNLReceiptCodeCalculator.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\Pdk\App\Order\Calculator\PostNl;
+
+use MyParcelNL\Pdk\App\Order\Calculator\AbstractPdkOrderOptionCalculator;
+use MyParcelNL\Pdk\Base\Service\CountryCodes;
+use MyParcelNL\Pdk\Types\Service\TriStateService;
+use MyParcelNL\Pdk\Validation\Validator\CarrierSchema;
+use MyParcelNL\Pdk\Facade\Pdk;
+
+/**
+ * When receipt code is enabled, insurance is required.
+ */
+final class PostNLReceiptCodeCalculator extends AbstractPdkOrderOptionCalculator
+{
+    public function calculate(): void
+    {
+        $shipmentOptions = $this->order->deliveryOptions->shipmentOptions;
+
+        if ($this->order->shippingAddress->cc !== CountryCodes::CC_NL) {
+            $shipmentOptions->receiptCode = TriStateService::DISABLED;
+            return;
+        }
+
+        if (TriStateService::ENABLED === $shipmentOptions->ageCheck) {
+            $shipmentOptions->receiptCode = TriStateService::DISABLED;
+            return;
+        }
+
+        if (TriStateService::ENABLED !== $shipmentOptions->receiptCode) {
+            return;
+        }
+
+        $shipmentOptions->signature     = TriStateService::DISABLED;
+        $shipmentOptions->onlyRecipient = TriStateService::DISABLED;
+        $shipmentOptions->largeFormat   = TriStateService::DISABLED;
+        $shipmentOptions->return        = TriStateService::DISABLED;
+
+        if ($shipmentOptions->insurance <= 1) {
```

Comment:
Als `insurance` een TriState waarde heeft (wat ik vermoed op basis van deze statement) dan lijkt het me netter om hem te schrijven als

```suggestion
        if ($shipmentOptions->insurance !== TriStateService::ENABLED) {
```

Als dit niet klopt en dit geen TriState is, negeer deze comment dan :)